src/components/dls/Modal/Modal.stories.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from 'react';
+
+import Modal from './Modal';
+
+export default {
+  title: 'dls|Modals',
+  component: Modal,
+  argTypes: {
+    visible: {
+      defaultValue: false,
+      description: `Whether the modal dialog is visible or not`,
+      table: {
+        category: 'Required',
+      },
+    },
+    centered: {
+      defaultValue: false,
+      description: `Whether the modal should be centered or not.`,
+      table: {
+        category: 'Optional',
+      },
+      options: [true, false],
+      control: { type: 'select' },
+    },
+    showCloseIcon: {
+      defaultValue: true,
+      description: `Whether a close (x) button is visible on the top right of the modal or not.`,
+      table: {
+        category: 'Optional',
+      },
+      options: [true, false],
+      control: { type: 'select' },
+    },
+    canCloseByKeyboard: {
+      defaultValue: true,
+      description: `Whether to support pressing esc to close the modal or not.`,
+      table: {
+        category: 'Optional',
+      },
+      options: [true, false],
+      control: { type: 'select' },
+    },
+    closeWhenClickingOutside: {
+      defaultValue: true,
+      description: `Whether to close the modal dialog when the mask (area outside the modal) is clicked.`,
+      table: {
+        category: 'Optional',
+      },
+      options: [true, false],
+      control: { type: 'select' },
+    },
+    children: {
+      description: 'The modal body',
+      table: {
+        category: 'Optional',
+      },
+    },
+    title: {
+      description: 'The title of the modal.',
+      table: {
+        category: 'Optional',
+      },
+    },
+    width: {
+      defaultValue: 520,
+      description: 'The width of the modal in pixels.',
+      table: {
+        category: 'Optional',
+      },
+      control: { type: 'number' },
+    },
+    closeIcon: {
```

Comment:
rename to `customCloseIcon` otherwise it might be understood as a boolean prop that takes true/false